Add ability to update Frigate+ model to latest from UI

leccelecce opened the pull request:
## Proposed change

* Adds an `/api/plus/models` endpoint which itself calls Frigate+'s `model/list` endpoint.
* Adds a dropdown to the Settings->Frigate+ page to show the available models for the current detector type and allow saving

![image](https://github.com/user-attachments/assets/6877c24b-1653-4ca4-8694-ebcedfd09ad2)

This borrows heavily from the Classification settings page. The React / UI design may need some review.

@blakeblackshear do you have any objection to the Frigate+ API being called when users navigate to the Frigate+ settings page? I'm assuming this is relatively infrequent and so won't generate excessive load on your infrastructure but if you would prefer it was a manual "Check for updates" link I could change to that.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code
- [ ] Documentation Update

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:

## Checklist

<!--
  Put an `x` in the boxes that apply.
-->

- [X] The code change is tested and works locally.
- [X] Local tests pass. **Your PR cannot be merged unless tests pass**
- [X] There is no commented out code in this PR.
- [X] UI changes including text have used i18n keys and have been added to the `en` locale.
- [X] The code has been formatted using Ruff (`ruff format frigate`)

leccelecce committed:
Add ability to update Frigate+ model to latest

hawkeye217 commented:
The full model ID seems like it's mostly unhelpful information (as long hash values typically are) to be something displayed so prominently. If anything, the first or last few characters is all I'm really looking at to confirm anything. Perhaps the dropdown should display the date/time of the model generation rather than the model ID, or perhaps just as smaller text in a div in the select box? Curious what the other maintainers think.

NickM-27 commented:
Yes, we also need to make sure we're showing what detector the model belongs to

leccelecce commented:
It currently filters to only show models compatible with the current detector config. I could change to show all available models, and just have incompatible ones disabled/greyed out.

hawkeye217 commented:
That seems reasonable.

It seems like we could also minimize the prominence of the model ID in the Model Information pane as well. I purposely left it out when originally coding it for the same above reasons. It's a lot easier to remember a date/time than even a few characters of a hash.

leccelecce committed:
UI tweaks

leccelecce commented:
Removed the model ID from the display section
Added timestamp to the dropdown with ID given reduced prominence.

![image](https://github.com/user-attachments/assets/9815fc78-5779-4700-84e3-e6990a6cd9ac)

hawkeye217 commented:
The select box's default should be the date/time as well, not the model ID. And it should follow the date format the rest of the UI uses (which is based on `config.ui.time_format`). I'd prefer to see the model ID in a smaller font below the date/time as well.

leccelecce commented:
The date format was to match the approach existing Train Date field. I've just had a look and can't see any existing functions for working off non-timestamp date fields, any pointers?

Updates so far - added base model, supported detectors, and now show all models but disable if wrong type/unsupported detector.

![image](https://github.com/user-attachments/assets/b4233e68-4961-4716-bd09-b389c6c6c3c0)

leccelecce committed:
further UI tweaks

NickM-27 commented:
Nice

hawkeye217 commented:
Looks good to me

leccelecce commented:
Thanks both for the feedback.

The only thing is I don't have multiple variants of models in my Frigate+ account and would need to figure out how to mock a response - so not fully tested from a all-config-variants perspective.

NickM-27 commented:
I can test

leccelecce commented:
Ah - just realised I've got a bug in the Select since changing to read the existing model config, but run out of time to fix. will leave as draft

hawkeye217 commented:
We'll want to include the `width` and `height` from the plus api response in the select dropdown as well, yolonas models are generated in 640x640 and 320x320, so there'd be no way to distinguish the two in the dropdown, currently. Might also be good to have that next to the model type in the Model Information pane, eg: "yolonas (320x320)"

NickM-27 commented:
<img width="716" alt="Screenshot 2025-03-23 at 5 49 54 PM" src="https://github.com/user-attachments/assets/684633bc-5a7a-40d1-80b0-77c6b35fb95a" />

leccelecce commented:
My problem is the SelectTrigger not visibly updating on a selection change even though the underlying value is changed, presumably I need to tie it to the actual selected model state rather than just read the config value...bit of rework

leccelecce committed:
UI tweaks: add width and height, fix select

leccelecce committed:
Add placeholder while API call in progress

leccelecce committed:
Fix Frigate+ enabled check

leccelecce commented:
Now looks like this, plus a few other background fixes:

![image](https://github.com/user-attachments/assets/311afa42-ffd9-46c8-afd8-f48705ad5f99)

leccelecce committed:
Fix config change lost when reloading page

leccelecce committed:
Add persistent message requiring restart

hawkeye217 commented:
Looking good, though this still needs a few more tweaks.

- The expanded select box overflows on mobile. Probably could just drop the type and size down on a new line.
- The selected model only shows the date/time. For me, because yolonas generates 640 and 320 variants and the timestamp is the same, I can't distinguish which model I have selected by only the timestamp. The select should probably just display the same things that the dropdown items display for clarity.

leccelecce committed:
Drop down supported detectors and dimensions

leccelecce committed:
Add width and height to display

NickM-27 commented:
looks like we need a space or `()`
![Screen Shot 2025-03-24 at 07 21 56 AM](https://github.com/user-attachments/assets/161e2f4f-ef06-4c18-a20a-21d59401a6d2)

leccelecce committed:
Update FrigatePlusSettingsView.tsx

hawkeye217 commented:
Still need to add i18n keys for a few of your new text additions.

leccelecce commented:
Ha, I'm out and about and been doing this on mobile / terminal, hence the flurry of commits. I can't figure out how to run the formatter but I've just discovered Codespaces...

leccelecce committed:
Temp fix for Codespaces not loading

leccelecce committed:
Add i18n, format

leccelecce committed:
remove unneeded brackets

leccelecce committed:
missing colon

leccelecce committed:
Revert "Temp fix for Codespaces not loading"

This reverts commit 75b19674ce3c33e69308358c29e80bf2774f377d.

leccelecce commented:
OK, this is finally done. I'm not sure why the Python / Devcontainer actions are failing, looks to be unrelated.

NickM-27 commented:
I will fix that separately. Looks related to imutils and the last commit was years ago, so we can probably just drop it anyway

leccelecce commented:
Great stuff. Many thanks to you both for the patience on this!
